### PR TITLE
[plugin.video.fsgo] 1.1.1

### PR DESCRIPTION
**Pull request by emilsvennesson**
### Description
This is a video add-on that allows users to access FS GO with their TV provider login.
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a sinlge commit with using the following style: [plugin.video.foo] v1.0.0

Addtional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuesly monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
@learningit @phil65 @enen92  ping for review

File: plugin.video.fsgo/default.py [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+    verify_ssl = False
+else:
+    verify_ssl = True
+
+fsgo = fsgolib(cookie_file, credentials_file, debug, verify_ssl)
+
+
+def addon_log(string):
+    if debug:
+        xbmc.log('%s: %s' % (logging_prefix, string))
```
Comment (enen92) [inline plugin.video.fsgo/default.py]: Please set the logging level to xbmc.LOGDEBUG as the default for Jarvis is still xbmc.LOGNOTICE (that will change for krypton and future versions). The rational is not to have the log file too spammed and to have a centralized way of enabling debugging

File: plugin.video.fsgo/resources/lib/fsgo.py [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+import cookielib
+import time
+from datetime import datetime, timedelta
+import calendar
+import uuid
+from urllib import urlencode
+
+import requests
+import m3u8
+import iso8601
```
Comment (enen92) [inline plugin.video.fsgo/resources/lib/fsgo.py]: You missed this one on addon.xml

Comment (enen92) [inline plugin.video.fsgo/resources/lib/fsgo.py]: This one is not defined in the addon.xml

File: plugin.video.fsgo/resources/lib/fsgo.py [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+        def __init__(self, value):
+            self.value = value
+
+        def __str__(self):
+            return repr(self.value)
+
+    def log(self, string):
+        if self.debug:
+            try:
+                print '[fsgolib]: %s' % string
```
Comment (enen92) [inline plugin.video.fsgo/resources/lib/fsgo.py]: Print statements are not allowed in addons, please use xbmc.log for that

File: plugin.video.fsgo/resources/lib/fsgo.py [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+            except UnicodeEncodeError:
+                # we can't anticipate everything in unicode they might throw at
+                # us, but we can handle a simple BOM
+                bom = unicode(codecs.BOM_UTF8, 'utf8')
+                print '[fsgolib]: %s' % string.replace(bom, '')
```
Comment (enen92) [inline plugin.video.fsgo/resources/lib/fsgo.py]: Same here

File: plugin.video.fsgo/resources/lib/fsgo.py [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+        """Register FS GO session. Write session_id and authentication header to file."""
+        url = self.base_url + '/sessions/registered'
+        session = {}
+        session['device'] = {}
+        session['location'] = {}
+        session['device']['token'] = self.get_credentials()['access_token']
+        session['device']['platform'] = 'ios-tablet'
+        session['location']['latitude'] = '0'  # unsure if this needs to be set
+        session['location']['longitude'] = '0'  # unsure if this needs to be set
+        post_data = json.JSONEncoder().encode(session)
```
Comment (enen92) [inline plugin.video.fsgo/resources/lib/fsgo.py]: You could probably just use json.dumps() here

File: plugin.video.fsgo/resources/lib/fsgo.py (hunk 3 above)
Comment (emilsvennesson) [inline plugin.video.fsgo/resources/lib/fsgo.py]: @enen92: This lib is supposed to be able to run independent of Kodi, which means that I can't import any xbmc-related modules in it. This has not been a problem for a few other add-ons I have submitted.

Comment (emilsvennesson) [inline plugin.video.fsgo/resources/lib/fsgo.py]: @enen92: Thanks!

**Comment by emilsvennesson** [comment]
@enen92 @phil65 This should be good to go now I think!

File: plugin.video.fsgo/resources/lib/fsgo.py [hunk 6]
```diff
@@ -0,0 +1,415 @@
...
+    def get_credentials(self):
+        try:
+            with open(self.credentials_file, 'r') as fh_credentials:
+                return json.loads(fh_credentials.read())
+        except IOError:
+            self.reset_credentials()
+            with open(self.credentials_file, 'r') as fh_credentials:
+                return json.loads(fh_credentials.read())
+
+    def heartbeat(self):
```
Comment (emilsvennesson) [inline plugin.video.fsgo/resources/lib/fsgo.py]: Whoops... this has been fixed.

**Pull request merged**